contracts/libraries/SwapIntention.sol

```diff
@@ -0,0 +1,75 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.20;
+
+import {Currency} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {IHooks} from "@uniswap/v4-core/contracts/interfaces/IHooks.sol";
+import {PoolKey} from "@uniswap/v4-core/contracts/types/PoolKey.sol";
+
+enum SwapType {
+    ExactInput,
+    ExactInputSingle,
+    ExactOutput,
+    ExactOutputSingle
+}
+
+struct SwapInfo {
```

Comment:
This one is pretty specific to Quoter, right? so should probably live in Quoter.sol

Reply:
same with SwapType enum